app/Controllers/CtrlSolution.php

```diff
@@ -20,114 +39,170 @@ public function viewSolutions()
 
     public function viewSolutionCreate()
     {
-        return view('admin/typeSolutions/SolutionCreate');
+        if (session()->has('clientData')) {
+            $this->fileConfig->setDataInClientConfig(session()->get('clientData'));
+        }
+
+        return view('admin/typeSolutions/SolutionCreate', ['filepondConfig' => $this->fileConfig->getClientConfig()]);
     }
 
-    public function viewSolutionEdit($id)
+    public function viewSolutionEdit($msId)
     {
-        $measurementSolutionModel = new MeasurementSolutionModel();
-        $measurementSolution      = $measurementSolutionModel->find($id);
+        $msData = (new MeasurementSolutionModel())->getMeasurementSolutionDataWithFiles($msId);
+
+        if (! session()->has('clientData')) {
+            $dataFiles['msIcon']  = [$msData['msIcon']];
+            $dataFiles['msImage'] = [$msData['msImage']];
+        } else {
+            $dataFiles = session()->get('clientData');
+        }
 
-        return view('admin/typeSolutions/SolutionEdit', ['solution' => $measurementSolution]);
+        $this->fileConfig->setDataInClientConfig($dataFiles);
+
+        return view('admin/typeSolutions/SolutionEdit', ['solution' => $msData, 'filepondConfig' => $this->fileConfig->getClientConfig()]);
     }
 
     public function createSolution()
     {
+        $msData = $this->request->getPost();
+
         try {
-            $newMeasurementSolutionData       = $this->request->getPost();
-            $measurementSolutionDataValidator = new SolutionValidation();
+            $msDataValidator = new SolutionValidation();
 
-            if (! $measurementSolutionDataValidator->validateInputs($newMeasurementSolutionData)) {
-                throw new InvalidInputException($measurementSolutionDataValidator->getErrors());
+            if (! $msDataValidator->validateInputs($msData)) {
+                throw new InvalidInputException($msDataValidator->getErrors());
```

Comment:
Here you are not validating the files. This could be like following:
```suggestion
            $fileValidationRules = $this->fileConfig->getCollectionFileValidationRules();
            $msDataValidator->addRules($fileValidationRules['rules'], $fileValidationRules['messages']);
            if (! $msDataValidator->validateInputs($msData)) {
                throw new InvalidInputException($msDataValidator->getErrors());
            }
```